extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/manager/impl/DeviceStatusManagerImpl.java

```diff
@@ -0,0 +1,937 @@
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.sensorJobExecuter.sensorJob.impl.DeviceConsumptionSensorJob;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.sensorJobExecuter.sensorJob.impl.DeviceOutputValueSensorJob;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.sensorJobExecuter.sensorJob.impl.SceneConfigSensorJob;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.sensorJobExecuter.sensorJob.impl.SceneOutputValueSensorJob;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.serverConnection.DsAPI;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.Device;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.CachedMeteringValue;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.DSID;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.DeviceConstants;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.DeviceStateUpdate;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.DeviceStateUpdateImpl;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.MeteringTypeEnum;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.MeteringUnitsEnum;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.OutputModeEnum;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.SensorEnum;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.scene.InternalScene;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.scene.constants.ApartmentSceneEnum;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.scene.constants.SceneEnum;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DeviceStatusManagerImpl} is the implementation of the the
+ * {@link DeviceStatusManager}.
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Matthias Siegele - Initial contribution
+ */
+public class DeviceStatusManagerImpl implements DeviceStatusManager {
+
+    private Logger logger = LoggerFactory.getLogger(DeviceStatusManagerImpl.class);
+
+    private ConnectionManager connMan;
+    private StructureManager strucMan;
+    private SceneManager sceneMan;
+    private DsAPI digitalSTROMClient;
+
+    private SensorJobExecutor sensorJobExecuter = null;
+    private SceneSensorJobExecutor sceneJobExecuter = null;
+
+    private static final int POLLING_FREQUENCY = Config.POLLING_FREQUENCY; // in seconds
+    private int BIN_CHECK_TIME = Config.BIN_CHECK_TIME; // in milliseconds
+
+    private Thread schedduler2 = null;
```

Comment:
What's about other threads of the digitalSTROM library like the event-listener. Shall we use the SingleThreadScheduledExecutor every time or is there more effective way? We found the ThreadPoolManager at the BaseThingHandler.
Is it a good idea to use it at the library? In that case, is there an existing pool we can use or shall we use a new pool named like "digitalstromLib"?
We think only the SensorJobExecutors will be difficult to change.